koku/api/query_params.py

```diff
@@ -416,23 +417,36 @@ def _set_tag_keys(self, query_params):
         self.tag_keys = param_tag_keys
 
     def _set_aws_category_keys(self, query_params):
-        """Set the valid aws_category keys"""
+        """Set the valid self.aws_category keys.
+
+        The aws_category_keys variable is used in the report serializer
+        to update the valid field names list. Any key added to this set
+        will not a trigger the unsupport parameter or invalid value error.
+        """
         prefix_list = [AWS_CATEGORY_PREFIX, AND_AWS_CATEGORY_PREFIX, OR_AWS_CATEGORY_PREFIX]
         self.aws_category_keys = set()
         if not any(f"[{prefix}" in self.url_data for prefix in prefix_list):
             return
-        param_aws_category_keys = set()
+        enabled_category_keys = set()
+        with tenant_context(self.tenant):
+            enabled_category_keys.update(
+                AWSEnabledCategoryKeys.objects.values_list("key", flat=True).filter(enabled=True).distinct()
+            )
+        if not enabled_category_keys:
+            return
+        # Make sure keys passed in exist in the DB.
         for key, value in query_params.items():
+            # Check key
+            stripped_key = self._strip_prefix(key, AWS_CATEGORY_PREFIX, prefix_list)
+            if stripped_key in enabled_category_keys:
+                self.aws_category_keys.add(stripped_key)
+            # Check Values
             if not isinstance(value, (dict, list)):
                 value = [value]
             for inner_key in value:
-                if AWS_CATEGORY_PREFIX in inner_key:
-                    param_aws_category_keys.add(inner_key)
-                    # TODO: Right now we accept any key passed in.
-                    # However, we should rewrite this to only
-                    # allow acceptable categorites keys based off
-                    # keys found in the DB. (like above)
-        self.aws_category_keys = param_aws_category_keys
+                stripped_key = self._strip_prefix(inner_key, AWS_CATEGORY_PREFIX, prefix_list)
+                if stripped_key in enabled_category_keys:
+                    self.aws_category_keys.add(inner_key)
```

Comment:
I feel like this could be reduced to a recursive function, but I've been messing around with it for an hour and haven't come up with a good solution. :shipit: